go/integTest/shared_commands_test.go

```diff
@@ -5208,3 +5209,198 @@ func (suite *GlideTestSuite) TestZScan() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+		var resultInt64 api.Result[int64]
+
+		resultInt64, err := client.SetBit(key, 3, 1)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), int64(0), resultInt64.Value())
+
+		resultInt64, err = client.SetBit(key, 5, 1)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), int64(0), resultInt64.Value())
+
+		result, err := client.Get(key)
+		assert.Nil(suite.T(), err)
+		value := result.Value()
+
+		binaryString := fmt.Sprintf("%08b", value[0])
+
+		assert.Equal(suite.T(), "00010100", binaryString)
+	})
+}
+
+func (suite *GlideTestSuite) TestWait() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+		client.Set(key, "test")
+		// Test 1:  numberOfReplicas (2)
+		resultInt64, err := client.Wait(2, 2000)
+		assert.Nil(suite.T(), err)
+		assert.True(suite.T(), resultInt64.Value() >= 2)
+
+		// Test 2: Invalid numberOfReplicas (0)
+		_, err = client.Wait(0, 2000)
+
+		// Assert error and message for invalid number of replicas
+		assert.NotNil(suite.T(), err)
+		assert.Equal(suite.T(), "number of replicas should be greater than 0", err.Error())
+
+		// Test 3: Invalid timeout (negative)
+		_, err = client.Wait(2, -1)
+
+		// Assert error and message for invalid timeout
+		assert.NotNil(suite.T(), err)
+		assert.Equal(suite.T(), "timeout cannot be lesser than 0", err.Error())
+	})
+}
+
+func (suite *GlideTestSuite) TestGetBit_ExistingKey_ValidOffset() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+		offset := int64(7)
+		value := int64(1)
+
+		client.SetBit(key, offset, value)
+
+		result, err := client.GetBit(key, offset)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), value, result.Value())
+	})
+}
+
+func (suite *GlideTestSuite) TestGetBit_NonExistentKey() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+		offset := int64(10)
+
+		result, err := client.GetBit(key, offset)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), int64(0), result.Value())
+	})
+}
+
+func (suite *GlideTestSuite) TestGetBit_InvalidOffset() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+		invalidOffset := int64(-1)
+
+		_, err := client.GetBit(key, invalidOffset)
+		assert.NotNil(suite.T(), err)
+	})
+}
+
+func (suite *GlideTestSuite) TestBitCount_ExistingKey() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+		for i := int64(0); i < 8; i++ {
+			client.SetBit(key, i, 1)
+		}
+
+		result, err := client.BitCount(key)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), int64(8), result.Value())
+	})
+}
+
+func (suite *GlideTestSuite) TestBitCount_ZeroBits() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := uuid.New().String()
+
+		result, err := client.BitCount(key)
+		assert.Nil(suite.T(), err)
```

Comment:
Nit: use `NoError` - it gives nicer report in case of failure
```suggestion
		assert.NoError(suite.T(), err)
```
Please update in all places in your PR

Reply:
done